src/parser.rs

```diff
@@ -1132,6 +1205,60 @@ mod tests {
     }
 
     #[test]
+    fn test_ast_macro_definition() {
+        let parsed_ast = parse("{% macro helloworld(greeting) %}Hello{% endmacro helloworld %}");
```

Comment:
There should probably be tests that use zero parameters and more than one parameter.